processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -330,25 +329,67 @@ public static Long convertObjectToLong(@Nullable Object valObj, boolean reportPa
     } else if (valObj instanceof String) {
       Long ret = DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
       if (reportParseExceptions && ret == null) {
-        throw new ParseException((String) valObj, "could not convert value [%s] to long", valObj);
+        final String message = (objectKey != null) ? StringUtils.nonStrictFormat(
+            "could not convert value [%s] to long for dimension [%s]",
+            valObj,
+            objectKey
+        ) : StringUtils.nonStrictFormat(
+            "could not convert value [%s] to long",
+            valObj
+        );
+        throw new ParseException((String) valObj, message);
       }
       return ret;
     } else if (valObj instanceof List) {
-      throw new ParseException(
-          valObj.getClass().toString(),
+      final String message = (objectKey != null) ? StringUtils.nonStrictFormat(
```

Comment:
Should we also include `"Invalid type: []` with `valObj.getClass()"` similar to the other error message? (Either instead of "cannot have multiple values" or as something that complements it to be clear)

Reply:
I dont know. I think i'd prefer to not touch that and keep this change smaller and to the point